helm-chart/binderhub/templates/_helpers.tpl

```diff
@@ -15,7 +15,32 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
-{{- define "imagePullSecret" }}
-{{- $authHost := default .Values.registry.host .Values.registry.authHost -}}
-{{- printf "{\"auths\": {\"%s\": {\"auth\": \"%s\"}}}" $authHost (printf "%s:%s" .Values.registry.username .Values.registry.password | b64enc) | b64enc }}
+{{/*
+Render docker config.json for the registry-publishing secret.
+*/}}
+{{- define "registryDockerConfig" -}}
+
+{{- /* default auth host */ -}}
+{{- $authHost := (default "https://index.docker.io/v1" .Values.registry.host) }}
```

Comment:
Since it is changing how these are set in a breaking way, now might be a good time to do it:

How about:

- auth_key: the key in docker config.json to find authentication data for the registry (typically `https://registry.host`)
- registry_url: the URL for the registry (typically `https://registry.host`), set from auth_key by default.

?

Reply:
On reading more about the docker login API, this should be a URL, too.

I also think that docker hub may be the *only* registry where the auth_url (the key in docker config) and registry url can differ.

So it might be something that could be simplified to not have two configurables here, and only special-case transforming `index.docker.io` -> `registry.hub.docker.com`.

Reply:
Reworked to be simplified to emphasize that these values ~never differ for user-specified registries.

The keys are now:

- Registry.url (the actual url of the registry)
- Registry.auth_config_url (the url used to find auth in docker config.json, only in the extremely unlikely situation that it differs from Registry.url)

Reply:
+1 to moving to URL instead of host since it's easier for users to comprehend.